File: array.cpp
#include<iostream>
using namespace std;
#define capacity 7
int sz=3;

void display(int arr[]){
	cout<<endl;
	for(int i=0;i<sz;i++){
		cout<<arr[i]<<" ";
	}
	cout<<endl;
}
void insert_at_end(int arr[],int x){
	arr[sz]=x;
	sz++;
	cout<<"Insert at end: ";
	display(arr);

}

void insert_at_beg(int arr[],int x){
	if(sz==0){
		arr[0]=x;
	}
	else{
		for(int i=sz;i>=0;i--){
			arr[i+1]=arr[i];
		}
		arr[0]=x;
	}
	sz++;
	cout<<"Insert at beginning: ";
	display(arr);
}
void insert_at_index(int arr[],int x,int index){
	if(index==0){
		insert_at_beg(arr,x);
	}
	else if(index==capacity){
		cout<<"array is full";
	}
	else if(index=sz){
		insert_at_end(arr,x);
	}
	else{
		for(int i=index;i<=sz;i++){
			arr[i+1]=arr[i];
		}
		arr[index]=x;
		sz++;
	}
	cout<<"Inserting at index: ";
	display(arr);
}
bool chkunderflow(){
	if(sz==0){
		cout<<"Underflow";
		return true;
	}
	return false;
}
void delete_at_beginning(int arr[]){
	if(chkunderflow()){
		return;
	}
	else{
		for(int i=0;i<sz;i++){
			arr[i]=arr[i+1];
		}
		sz--;
	}
	cout<<"After deleting at beginning: ";
	display(arr);
}
void delete_at_end(int arr[]){
	if(chkunderflow()){
		return;
	}
	else{
		sz--;
	}
	cout<<"After deleting at end: ";
	display(arr);
}
void delete_at_pos(int arr[],int pos){
	if(chkunderflow()){
		return;
	}
	else{
		for(int i=pos;i<sz;i++){
			arr[i]=arr[i+1];
		}
		sz--;
	}
	cout<<"After deleting at position: ";
	display(arr);
}
int main(){
	int a[capacity];
	a[0]=1;
	a[1]=2;
	a[2]=3;
	delete_at_pos(a,1);
	delete_at_end(a);
	delete_at_beginning(a);
	delete_at_end(a);
}

File: Array/Array.java
public class Array {
    int capacity=7;
    int size = 0;
    public static void main(String[] args) {
        int[] arr = new int[7];
        Array obj = new Array();
        obj.Insertatbeg(arr,4);
        obj.Insertatbeg(arr,5);
        obj.Display(arr);
        obj.Deletionatend(arr);
        obj.Display(arr);
        obj.Deletionatindex(arr,1);
        obj.Display(arr);
    }
    void Insertatbeg(int[] arr, int x){
        if(size==0){
            arr[0]=x;
            size++;
            return;
        }
        for(int i=size-1;i>=0;i--){
            arr[i+1]=arr[i];
        }
        arr[0]=x;
        size++;
    }
    void Insertatend(int[] arr, int x){
        if(size>=capacity){
            System.out.println("array is full");
            return;
        }
        arr[size++]=x;
        size++;
    }
    void Insertatindex(int[] arr, int x, int index){
        if(index==0){
            Insertatbeg(arr,x);
            return;
        }
        if(index==size-1){
            Insertatend(arr,x);
            return;
        }
        for(int i=size-1; i>=index;i--){
            arr[i+1]=arr[i];
        }
        arr[index]=x;
        size++;
    }
    void Deletionatend(int[] arr){
        if(size==0){
            System.out.println("underflowwwwww");
            return;
        }
        size--;
    }
    void Deletionatbeg(int[] arr){
        if(size==0){
            System.out.println("underflowwwwww");
            return;
        }
        for(int i=0;i<=size-1;i++){
            arr[i]=arr[i+1];
        }
        size--;
    }
    void Deletionatindex(int[] arr,int index){
        if(size==0){
            System.out.println("underflowwwwww");
            return;
        }
        if(index>=size-1){
            Deletionatend(arr);
            return;
        }
        for(int i=index; i<=size-1;i++){
            arr[i]=arr[i+1];
        }
        size--;
    }
    void Display(int[] arr){
        System.out.println();
        for(int i = 0; i<size; i++) {
            System.out.print(arr[i]);
        }
        System.out.println();
    }
}